### ISSUE TITLE
Status Issue - please read before PR or Merge (updated Apr 21)

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#3677]
From #announcements in discord on Feb 6:

-----
Hi folks! Just a quick update on Our plans for Surge 1.9 and beyond.

1.8.1 seems to have stuck and so we are in the process of adding features and changing aspects of the Surge architecture. This leads us on a two pronged approach for the first half of 2021 (if everything works)

1. This spring - probably April - we will release Surge 1.9 which will contain a bunch of new effects, maybe a new oscillator or two, a set of quality of life improvements, and general neat things. But most critically we will not expand the parameter space of Surge with this release. The next nightly of Surge will be labeled "1.9" and https://github.com/surge-synthesizer/surge/milestone/20 are the tentative issues.

2. In parallel with that effort we are working on porting Surge entirely to JUCE in a product we call Surge XT. That will be a new plugin ID with the same core engine, but with architectural changes to make it much easy for us to change parameters, GUI, and more. We plan the first release of Surge XT in summer 2021, and plan to have collaborator pre-alpha releases available as binaries in February. (You can build it now from source if you want but we don't have binary nightlies). If you are interested in Surge XT, please hang out in #surge-development for now. The book of work for the port to juce is in issue #3737

Finally, we are still chipping away on ShortCircuit slowly but surely. 

And so that's a lot of dev work. If you are reading this and thinking "hey I know C++, how can I help" please do get in touch in #surge-development or #sc-development.

------

How does that translate to dev work today? Well SurgeXT will eliminate loads of code so be careful where you invest your time.

1. If you are at or earlier than the surge-shared link target, everything is fine. Add FX, modify DSP, add tests to headless, all those sorts of things
2. If you want to work on the wrappers (vst3/ au/ lv2/ or vst2/) you are probably wasting your time. They will go away with juce and we will keep them as they are for 1.9
3. If you want to work on the GUI be very careful. Our JUCE strategy allows us to run VSTGUI-API code in JUCE but there's some things which could be wrong. If you are planning on doing GUI work of any form please hop on discord and chat with @baconpaul or open an issue before you begin.
4. The SurgeXT alpha builds in pipelines now so your PRs have to work with it. If they don't and you want to build locally check out the "Port to Juce" issue on how to activate the build in your local environment.

So basically: If you are writing any code upstream from the surge-shared target in CMakeLists.txt you need to coordinate with bp on the juce conflicts please!

At some point this spring, after the 1.9 release, we will fully commit to juce and remove src/vst[23], src/au, src/lv2 vstgui.surge and vst3sdk from the codebase. At that point we will also add juce as a submodule (right now it checks out on demand from cmake). More as that proceeds.
